src/select/custom-select/custom-select.component.ts

```diff
@@ -330,6 +318,21 @@ export class ThySelectCustomComponent implements ControlValueAccessor, IThyOptio
             this.instanceSelectionModel();
         }
         this.setDropDownClass();
+
+        if (isPlatformBrowser(this.platformId)) {
+            this.ngZone.runOutsideAngular(() =>
+                fromEvent(document, 'click')
```

Comment:
we support a `ThyClickDispatcher` service in core, it is recommended to use it subscribe document click event, so only one event will be bound on the document.

```ts
(private clickDispatcher: ThyClickDispatcher) {}

clickDispatcher.clicked(0)..pipe(takeUntil(this.destroy$)).subscribe((event)=>{ ... })

```

Reply:
Thanks, done.